lib/events.js

```diff
@@ -1017,80 +1010,15 @@ function eventTargetAgnosticAddListener(emitter, name, listener, flags) {
  * @returns {AsyncIterator}
  */
 function on(emitter, event, options) {
+  const queue = [];
```

Comment:
Queues are pretty slow for this sort of thing which is why the other one that looked at performance is using a FixedQueue

Reply:
Fixed